### PR TITLE
Limit usage of volatile storage

### PR DESCRIPTION
**Pull request by sosthene-nitrokey**
This PR limit the number of private RSA key in volatile storage to 1. 

See https://github.com/Nitrokey/nitrokey-3-firmware/issues/283#issuecomment-1587509088 for more context.

### CODE REVIEW

File: src/state.rs [hunk 1]
```diff
@@ -509,6 +509,76 @@ impl<'a> LoadedState<'a> {
...
+    /// Returns the requested key
+    pub fn key_id(
+        &mut self,
+        client: &mut impl crate::card::Client,
+        key: KeyType,
+        storage: Location,
+    ) -> Result<KeyId, Status> {
+        use KeyType as K;
+        use UserVerifiedInner as V;
+        // Self::clear_cached is there to avoid having multiple keys in the volatile storage.
```
Comment (robin-nitrokey) [inline src/state.rs:536]: outdated comment?

**Review by robin-nitrokey — CHANGES_REQUESTED** [review]

File: src/state.rs (hunk 1 above)
Comment (sosthene-nitrokey) [reply src/state.rs]: Indeed, the method was initially called `clear_cache`, I modified the comment to `limit_cache_size`, I still think the comment is relevant.

**Comment by sosthene-nitrokey** [comment]
Why didn't the CI run?

**Comment by robin-nitrokey** [comment]
AFAIS the CI ran but the checkmark is missing.

**Pull request merged**